server/src/main/java/org/apache/iotdb/db/mpp/sql/planner/LogicalPlanner.java

```diff
@@ -76,8 +84,155 @@ public LogicalPlanVisitor(Analysis analysis) {
 
     @Override
     public PlanNode visitQuery(QueryStatement queryStatement, MPPQueryContext context) {
-      // TODO: Generate logical planNode tree for query statement
-      return null;
+      PlanBuilder planBuilder = planSelectComponent(queryStatement);
+
+      if (queryStatement.getWhereCondition() != null) {
+        planBuilder =
+            planQueryFilter(planBuilder, queryStatement.getWhereCondition().getQueryFilter());
+      }
+
+      if (queryStatement.isGroupByLevel()) {
+        planBuilder =
+            planGroupByLevel(
+                planBuilder,
+                ((AggregationQueryStatement) queryStatement).getGroupByLevelComponent());
+      }
+
+      if (queryStatement instanceof FillQueryStatement) {
+        planBuilder =
+            planFill(planBuilder, ((FillQueryStatement) queryStatement).getFillComponent());
+      }
+
+      planBuilder = planFilterNull(planBuilder, queryStatement.getFilterNullComponent());
+      planBuilder = planSort(planBuilder, queryStatement.getResultOrder());
+      planBuilder = planLimit(planBuilder, queryStatement.getRowLimit());
+      planBuilder = planOffset(planBuilder, queryStatement.getRowOffset());
+      return planBuilder.getRoot();
+    }
+
+    private PlanBuilder planSelectComponent(QueryStatement queryStatement) {
+      Map<String, Set<SourceNode>> deviceNameToSourceNodesMap = new HashMap<>();
+      Map<String, DataRegionReplicaSet> deviceNameToDataRegionReplicaSetMap = new HashMap<>();
+
+      for (ResultColumn resultColumn : queryStatement.getSelectComponent().getResultColumns()) {
+        Set<SourceNode> sourceNodes = planResultColumn(resultColumn);
+        for (SourceNode sourceNode : sourceNodes) {
+          String deviceName = sourceNode.getDeviceName();
+          deviceNameToSourceNodesMap
+              .computeIfAbsent(deviceName, k -> new HashSet<>())
+              .add(sourceNode);
+          deviceNameToDataRegionReplicaSetMap.computeIfAbsent(
+              deviceName,
+              k -> analysis.getDataPartitionInfo().getDataRegionReplicaSet(k, null).get(0));
+          sourceNode.setDataRegionReplicaSet(deviceNameToDataRegionReplicaSetMap.get(deviceName));
```

Comment:
Is it necessary to set DataRegionReplicaSet to SourceNode here ?

And... If there are several `DataRegionReplicaSet`s here, we only set one of them to the SourceNode, which is not correct

Reply:
Got it ~ I will remove it